gateway/api/schedule.py

```diff
@@ -192,3 +194,15 @@ def handle_job_status_not_available(job: Job, job_status):
         job_status = Job.FAILED
         job.logs = f"{job.logs}\nSomething went wrong during updating job status."
     return job_status
+
+
+def send_notifications(job: Job):
+    slackToken = os.environ.get("SLACK_TOKEN")
+    slackChannel = os.environ.get("SLACK_CHANNEL")
+    if slackToken and slackChannel:
```

Comment:
Would it be possible to move this logic out of the function? Right now, we'd always call this function and then check if slack output was enabled. It might be better to check if slack output was enabled and if so then call this function...

Reply:
I want to keep the caller(handle in update_jobs_statuses.py) of this function simple.  I've just refactor it to simplify to remove the too_many_branches.  This function is called only then the job status is changed.  I guess it's not too bad for the performance point of view.  WDYT?

Reply:
I was thinking something like this:

```python
def handle():
    ...
    if slack_has_been_enabled: # however that's best determined
        send_notifications(job)
```

mainly to help with readability: if we haven't enabled slack, then there's no reason to call the notifications method. 

That said, I'm not super-familiar with the too-many-branches error, would adding an sometimes call of the notifications function cause that problem here?

Reply:
It's the total number of if, while, try and etc.. that cause the code branches. So adding `if` is not good from the point of view.

Reply:
That's a weird limitation of the framework, but I'm sure there's a reason for that :shrug: 

May also be another argument in favor of pushing alerts to the monitoring platform...

Reply:
Yes, this is one way to send the notification / alert.  This way is quite simple in our side.  So it' not bad to have the capability.